src/View/View.php

```diff
@@ -1338,17 +1338,20 @@ public function pluginSplit(string $name, bool $fallback = true): array
      * Returns layout filename for this template as a string.
      *
      * @param string|null $name The name of the layout to find.
-     * @return string Filename for layout file.
+     * @return string|null Filename for layout file.
      * @throws \Cake\View\Exception\MissingLayoutException when a layout cannot be located
      */
-    protected function _getLayoutFileName(?string $name = null): string
+    protected function _getLayoutFileName(?string $name = null): ?string
     {
         if ($name === null) {
-            /** @var string $name */
             $name = $this->layout;
         }
-        $subDir = '';
 
+        if (empty($name)) {
```

Comment:
Wouldnt the exception be better? Check line 985 ( $parent = $this->_getLayoutFileName($name); ) where it later tries to then compare false to a non false value and also tries to find an array value to this as key.

Reply:
So the return value of this should actually never be null IMO. But throw a meaningful exception why the input then was wrong here.

Reply:
No, if `View::$layout` is set to `false` means you don't want a layout to be rendered. That's not an exception case.

Reply:
Practically `renderLayout()` is never called if `View::$layout` is set to false.

Reply:
Exactly my point: No need to then make this nullable return if the input should not happen and should throw an exception instead to show a developer error instead of hiding this to blow up later in code.

Reply:
Could be InvalidArgumentException, LogicException or RuntimeException.

Reply:
And yet `renderLayout()` does check the return value `_getLayoutFileName()` :slightly_smiling_face: and bails out if it's empty, that's why I added the `null` return.

Reply:
not extend() and what I just explained there happening in code.
Also not, if you extended/replaced the code and badly order/code the execution chain of these protected methods.
Thus: better safe than sry IMO. And with a clear exception message. The in_array check with false as key does not seem safe/sane to me.

Reply:
`extend()` is always called with a layout name which is passed to `_getLayoutFileName()` hence `View::$layout` is never used in that case.

I really don't spend more time on this. With current code there's no code path which generates an error in actual usage. Feel free to make changes you see fit later on if you feel strongly about it.

Reply:
I just dont see the need or the intention of making _getLayoutFileName() nullable return.
We shouldnt weaken the API for no reason, that's all.

Reply:
I am fine cleaning this up with a follow up commit if I may.

Reply:
I have reverted the changes to `_getLayoutFileName()`.

Reply:
I am a bit how this discussion and the resolution went.